indexer/database/db.go

```diff
@@ -23,36 +26,41 @@ type DB struct {
 }
 
 func NewDB(dbConfig config.DBConfig) (*DB, error) {
-	dsn := fmt.Sprintf("host=%s port=%d dbname=%s sslmode=disable", dbConfig.Host, dbConfig.Port, dbConfig.Name)
-	if dbConfig.User != "" {
-		dsn += fmt.Sprintf(" user=%s", dbConfig.User)
-	}
-	if dbConfig.Password != "" {
-		dsn += fmt.Sprintf(" password=%s", dbConfig.Password)
-	}
-	gorm, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
-		// The indexer will explicitly manage the transaction
-		// flow processing blocks
-		SkipDefaultTransaction: true,
+	var db *DB
+
+	retryStrategy := &retry.ExponentialStrategy{Min: 1000, Max: 20_000, MaxJitter: 250}
+
+	_, err := retry.Do[interface{}](context.Background(), 10, retryStrategy, func() (interface{}, error) {
+		dsn := fmt.Sprintf("host=%s port=%d dbname=%s sslmode=disable", dbConfig.Host, dbConfig.Port, dbConfig.Name)
+		if dbConfig.User != "" {
+			dsn += fmt.Sprintf(" user=%s", dbConfig.User)
+		}
+		if dbConfig.Password != "" {
+			dsn += fmt.Sprintf(" password=%s", dbConfig.Password)
+		}
+		gorm, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
```

Comment:
nit: can also pull out the `gorm.Config` struct into a variable outside of the retry loop as well